core/src/apps/solana/transaction/instructions.py.mako

```diff
@@ -190,6 +198,7 @@ def get_instruction(
                 "${program["name"]}: ${instruction["name"]}",
                 True,
                 True,
+                ${getOptionalBool(instruction, "is_ui_hidden")},
```

Comment:
```suggestion
                ${instruction.get("is_ui_hidden", False)},
```

Reply:
same thing for `getOptionalString` btw. Mako templates are mostly just Python. that should have been just `ui_property.get("parameter")`

Reply:
For the bool i fixed it in cfb3bba , 
but for string there is an issue that if the when we are generating and the optional field contains some string we would like to have it returned as `"${ui_property.get("parameter")}"`, but when it does not we would like to have it returned as `${ui_property.get("parameter")}`

I can provide a quick example if needed as I am not really sure If i illustrated it well enough

Reply:
For the same functionality the lines with `getOptionalString` could be replaced with 
```
${'"'+ui_property.get("parameter")+'"' if "parameter" in ui_property else None},
```
which I don't think is realy nicely readable

Reply:
no, that's literally what `ui_property.get("parameter")` does

```
>>> foo = { "hello": 123 }
>>> print(foo.get("hello"))
123
>>> print(foo.get("world"))
None
```

Reply:
oh, i see, the problem is with the quotes.
well, okay, let's leave it as-is.

for the record you don't have to use Mako's syntax, you can just write a python function
this one is better btw:
```
def getOptionalString(obj, string):
    if string in obj:
        return json.dumps(obj[string])
    else:
        return "None"
```
json.dumps will take care of quotes and `\n`s and the like for you